### PR TITLE
#167190394 Implement Password Reset Functionality

### PR DESCRIPTION
**Pull request by Hadeneekeh**
#### What does this PR do?
Implement Password Reset Functionality

#### How should this be manually tested?
- Clone the repo
- Run the command below in your terminal to install all dependencies
```
npm install
``` 
- Run the command below to test
```
npm test
```

#### Any background context you want to provide?
[E-2-E-test-link](https://drive.google.com/file/d/16vocJyBysdTItDj4O-ec7tMjM84cMu4n/view?usp=sharing)

#### What are the relevant pivotal tracker stories?
- [#167190394](https://www.pivotaltracker.com/story/show/167190394)
#### Questions:
N/A

### CODE REVIEW

File: src/actions/__tests__/authActions.spec.js [hunk 1]
```diff
@@ -125,4 +140,104 @@ describe('Auth action tests', () => {
...
+    //   nock(url)
+    //     .post('/users/passwordReset')
+    //     .reply(400, errorResponse.err);
+
+    //   return store.dispatch(requestPasswordLink({}))
+    //     .then(() => {
+    //       expect(store.getActions()).toMatchSnapshot();
+    //     });
+    // });
+  });
```
Comment (topseySuave) [inline src/actions/__tests__/authActions.spec.js:188]: why do you have commented out codes? @Hadeneekeh

File: src/e2e/SignupTest.js [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
+//       .pause(1000)
+//       .setValue('#password', 'Password1$')
+//       .pause(1000)
+//       .setValue('#passwordConfirm', 'Password1$')
+//       .pause(1000)
+//       .click('[datatest="signup-submit"]')
+//       .saveScreenshot('tests_output/edit.png')
+//       .end();
+//   },
+// };
```
Comment (topseySuave) [inline src/e2e/SignupTest.js]: why?

File: src/e2e/profileTest.js [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+//       .assert.visible('input[type=file]')
+//       .setValue('input[type=file]', '/Users/igwechinonso/Downloads/event.jpeg')
+//       .pause(7000)
+//       .assert.visible('[datatest="submit-button"]')
+//       .click('[datatest="submit-button"]')
+//       .saveScreenshot('tests_output/edit.png')
+//       .pause(5000)
+//       .end();
+//   },
+// };
```
Comment (topseySuave) [inline src/e2e/profileTest.js]: ??

**Review by topseySuave — CHANGES_REQUESTED** [review]

**Comment by topseySuave** [comment]
@Hadeneekeh where is the E2E video?

File: src/actions/__tests__/authActions.spec.js (hunk 1 above)
Comment (Hadeneekeh) [reply src/actions/__tests__/authActions.spec.js:188]: I will fix it now

**Comment by Hadeneekeh** [comment]
> @Hadeneekeh where is the E2E video?

I have added it to the PR description.

**Review by chiboycalix — APPROVED** [review]
This looks good to me. Nice work @Hadeneekeh

**Review by encodedBicoding — APPROVED** [review]
Great job @Hadeneekeh. LGTM

**Review by obayomi96 — APPROVED** [review]
LGTM

File: src/actions/__tests__/authActions.spec.js (hunk 1 above)
Comment (Hadeneekeh) [reply src/actions/__tests__/authActions.spec.js]: Fixed

File: src/views/ResetPassword/ResetPassword.scss [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+
+        form {
+            margin-top: 50%;
+            transform: translateY(-50%);
+        }
+    }
+}
```
Comment (chuxmykel) [inline src/views/ResetPassword/ResetPassword.scss:30]: Great job @Hadeneekeh. I love your implementation and it works really well as I have been able to change my password and still log in successfully. Please add a new line to the end of this file and you're good to go.

**Review by chuxmykel — APPROVED** [review]
Looks good to me.

**Review by topseySuave — APPROVED** [review]

**Pull request merged**